examples/advanced/example_model_optimization.py

```diff
@@ -277,16 +279,24 @@ def enc_to_repr(self, enc: torch.Tensor) -> torch.Tensor:
 #   If you do not specify an ``optimizer``, the
 #   :func:`~pystiche.optim.default_model_optimizer`, i.e. :class:`~torch.optim.Adam` is
 #   used.
+#
+# .. note::
+#
+#   The weights of the provided transformer were trained with the
+#   `2014 training images <http://images.cocodataset.org/zips/train2014.zip>`_ of the
+#   `COCO dataset <https://cocodataset.org/>`_. The training was performed for
+#   ``num_epochs=2`` and ``batch_size=4``. Each image was center-cropped to
+#   ``256 x 256`` pixels.
 
 
-def train(
-    transformer, dataset, batch_size=4, epochs=2,
-):
-    if dataset is None:
-        raise RuntimeError(
-            "You forgot to define a dataset. For example, "
-            "you can use any image dataset from torchvision.datasets."
-        )
+def train(transformer, root, batch_size=4, epochs=2, image_size=256):
+    if root is None:
+        raise RuntimeError("You forgot to define a root image directory.")
+
+    image_transforms = nn.Sequential(
+        transforms.Resize(image_size), transforms.CenterCrop(image_size)
+    )
+    dataset = ImageFolderDataset(root, transform=image_transforms)
```

Comment:
Let's just keep it as `transform`:

```suggestion
    transform = nn.Sequential(
        transforms.Resize(image_size), transforms.CenterCrop(image_size)
    )
    dataset = ImageFolderDataset(root, transform=transform)
```